### ISSUE TITLE
TypeScript 4.8 error with strictNullChecks

### ISSUE THREAD

**Issue by claviustales** [recharts/recharts#3079]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/recharts-typescript-4-8-error-strictnullchecks-81s1f4)

### Steps to reproduce
I tried to reproduce, but I couldn't. Sorry.

### What is expected?
Compilation with no errors.

### What is actually happening?
Compilation shows the following error:

```
node_modules/recharts/types/component/LabelList.d.ts:21:70 - error TS2344: Type 'T' does not satisfy the constraint 'Data'.

21 export declare type Props<T> = SVGProps<SVGElement> & LabelListProps<T>;
                                                                        ~

  node_modules/recharts/types/component/LabelList.d.ts:21:27
    21 export declare type Props<T> = SVGProps<SVGElement> & LabelListProps<T>;
                                 ~
    This type parameter might need an `extends Data` constraint.
```

| Environment | Info |
|---|---|
| Recharts | v2.1.6 |
| React | 17.0.2 |
| System | MacOS 12.6.1 |
| Browser | Chrome 108.0.5359.94 |

**Comment by ckifer** [comment]
This should be fixed with the latest release 2.2.0
https://github.com/recharts/recharts/commit/65d4ee488e4a47da1f235e840e8140f091fb95b4

**Comment by ckifer** [comment]
Closing for now, let us know if this persists
